### DISCUSSION TITLE
Constantly cycling between Charging/Discharging - that expected?

### DISCUSSION THREAD

**Discussion by Allistah** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#164] [General]
I have a 2015 15" MacBook Pro.  It is working in general, but I would think that once it hits the desired charge point, it would then just run off the charger for power.  I use Coconut Battery to watch the state of the machine and I'm seeing that it is cycling between charging and discharging - staying right close to my set point of 50%.  Is that expected?  Since the PSU can supply enough power to power the machine on its own without the battery, why is it cycling like that?

Pretty cool tool - thank you!

**Comment by vassudanagunta** [comment]
> Since the PSU can supply enough power to power the machine on its own without the battery

I don't think that's true.

> It is well-known that any MacBook with battery removed will run at significantly lower clock speed, typically around 1 GHz. The reason is that the power supply is good enough to supply plenty of power for the average usage, but not for bursts. If your hard drive spins up while the CD drive spins up while all cores and the GPU work full blast and you are charging two iPhones then the power supply won't have enough power for that, but it can take extra power from the battery.

~ From https://apple.stackexchange.com/a/332246/263076

> Yeah, that's designed behavior. When both the CPU and GPU are being used at full tilt, it can draw more power than the power brick can provide, so it's designed to dip into the battery in these situations.

~ https://arstechnica.com/civis/viewtopic.php?f=19&t=1138558

More:

https://apple.stackexchange.com/a/401603/263076

https://www.quora.com/Will-a-MacBook-Pro-run-without-a-battery

**Reply by Allistah** [reply]
I flipped over to "Use Classic SMC Key", rebooted, and it seems to be more stable now.  The light on the MagSafe cord doesn't cycle back/forth between green/orange every second or so.  It seems to be staying orange now and doesn't go between running off the battery and running off the power adapter like it was before.  I'm running v2.1 alpha.  I'll leave it here for a bit and see if I get better results in the long run with this mode.

Another thing I noticed with the other mode was that it would suddenly start charging past the set point and I would come back to the machine and it would be 100% again.  I'll see if that happens with using the Classic SMC Key and report back.

**Reply by Allistah** [reply]
So far after the last hour, it's been rock solid on the Classic SMC Key method.  It has stayed at 51% exactly.  Before it would cycle between 49 and 52 - up to 52, down to 49, up to 51, down to 50.  SMC Key method seems to be the way to go for me on this laptop at least.  I'll give it overnight and see if I notice anything else and will report back tomorrow.

**Reply by Allistah** [reply]
Next day using Classic SMC Key method:  The LED on the MagSafe cord stays orange.  The power adapter stays listed as "Connected" and Charging with stays at "0 Watts".  The SOC (State of Charge) seems to stay right at 51% and Big Sur reports the Power Source is: Power Adapter, Battery is Not Charging.

For me, this is where I'll keep it I think.  Unless a new version comes out that someone would like me to test out.  I would think that the new method would aim to be a little more like what I described above not sure.
